src/ucp/rndv/rndv.c

```diff
@@ -650,33 +690,81 @@ static void ucp_rndv_req_init_get_zcopy_lane_map(ucp_request_t *rndv_req)
+                                                               dst_md_index);
+            lane_map                         |= UCS_BIT(i);
+            max_lane_bw                       = ucs_max(max_lane_bw, lane_bw);
         }
     }
 
     if (ucs_popcount(lane_map) > 1) {
         /* remove lanes if bandwidth is too less compare to best lane */
         ucs_for_each_bit(lane_idx, lane_map) {
             ucs_assert(lane_idx < UCP_MAX_LANES);
-            lane       = ep_config->rndv.get_zcopy_lanes[lane_idx];
+            lane       = lanes[lane_idx];
             rsc_index  = ep_config->key.lanes[lane].rsc_index;
             iface_attr = ucp_worker_iface_get_attr(ep->worker, rsc_index);
             lane_bw    = ucp_tl_iface_bandwidth(context, &iface_attr->bandwidth);
 
-            if ((lane_bw/max_lane_bw) <
+            if ((lane_bw / max_lane_bw) <
                 (1. / context->config.ext.multi_lane_max_ratio)) {
-                lane_map                                    &= ~UCS_BIT(lane_idx);
-                rndv_req->send.rndv_get.rkey_index[lane_idx] = UCP_NULL_RESOURCE;
+                lane_map                                &= ~UCS_BIT(lane_idx);
+                rndv_req->send.rndv.rkey_index[lane_idx] = UCP_NULL_RESOURCE;
             }
         }
     }
 
-    rndv_req->send.lanes_map_avail          =
-    rndv_req->send.rndv_get.lanes_map_all   = lane_map;
-    rndv_req->send.rndv_get.lanes_count     = ucs_popcount(lane_map);
+    ucp_rndv_req_init_lanes(rndv_req, lane_map, ucs_popcount(lane_map));
+}
+
+static void ucp_rndv_req_init(ucp_request_t *req, ucp_request_t *super_req,
+                              ucp_lane_map_t lanes_map, uint8_t lanes_count,
+                              ucp_rkey_h rkey, uint64_t remote_address,
+                              uint8_t *rkey_index)
+{
+    ucp_lane_index_t i;
+
+    req->super_req                = super_req;
+    req->send.rndv.rkey           = rkey;
+    req->send.rndv.remote_address = remote_address;
+
+    ucp_rndv_req_init_lanes(req, lanes_map, lanes_count);
+
+    if (rkey_index != NULL) {
+        memcpy(req->send.rndv.rkey_index, rkey_index,
+               sizeof(*req->send.rndv.rkey_index) * UCP_MAX_LANES);
+    } else {
+        for (i = 0; i < UCP_MAX_LANES; i++) {
+            req->send.rndv.rkey_index[i] = UCP_NULL_RESOURCE;
+        }
+    }
+}
+
+static void ucp_rndv_req_init_from_super_req(ucp_request_t *req,
+                                             ucp_request_t *super_req,
+                                             size_t length,
+                                             size_t send_buffer_offset,
+                                             size_t remote_address_offset,
+                                             ucs_ptr_map_key_t remote_req_id)
+{
+    req->super_req = super_req;
+    req->flags     = 0;
+    req->send.ep   = super_req->send.ep;
+
+    if (length != 0) {
```

Comment:
why?

Reply:
it was needed to use this function in `ucp_rndv_put_pipeline_frag_get_completion()`
but analyzing the code, we did extra actions, i.e. using `ucp_rndv_req_init_from_super_req()` is excessive there.
we only need to set `freq->send.rndv.remote_address`
so, removed using `ucp_rndv_req_init_from_super_req()` here, but added setting remote address.